### PR TITLE
Simplify output in login and project

### PR DESCRIPTION
**Pull request by smarterclayton**
@deads2k @fabianofranz

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
Did I miss the change for display name annotation key on server or are you just future proofing?  It was on my TODO to fix up tomorrow.

**Comment by smarterclayton** [comment]
Future proofing

File: pkg/cmd/cli/cmd/project.go [hunk 1]
```diff
@@ -168,13 +169,25 @@ func (o ProjectOptions) RunProject() error {
...
+				}
 
 				projects, err := getProjects(o.Client)
 				if err == nil {
-					msg += "\nYour projects are:"
-					for _, project := range projects {
-						msg += "\n" + project.Name
+					switch len(projects) {
+					case 0:
+						msg += "\nYou are not a member of any projects. You can request a project to be created with the 'new-project' command."
```
Comment (deads2k) [inline pkg/cmd/cli/cmd/project.go]: `osc new-project` ?

File: pkg/cmd/cli/cmd/project.go [hunk 2]
```diff
@@ -155,6 +155,7 @@ func (o ProjectOptions) RunProject() error {
 
 	contextInUse := ""
 	namespaceInUse := ""
+	isDefaultContext := false
```
Comment (deads2k) [inline pkg/cmd/cli/cmd/project.go]: This is badly named.  I'm trying to figure out what it is.  Is it marking the fact that no matching context was found and the name was treated as a project name?

File: pkg/cmd/cli/cmd/project.go [hunk 3]
```diff
@@ -199,13 +212,14 @@ func (o ProjectOptions) RunProject() error {
...
 		contextInUse = merged.CurrentContext
+		isDefaultContext = true
 	}
 
 	if err := kubecmdconfig.ModifyConfig(o.PathOptions, config); err != nil {
 		return err
 	}
 
-	if contextInUse != namespaceInUse {
+	if contextInUse != namespaceInUse && !isDefaultContext {
```
Comment (deads2k) [inline pkg/cmd/cli/cmd/project.go]: Why not unconditionally output, "Now using project %s on server %s as user %s"?

**Comment by deads2k** [comment]
minor comments.

File: pkg/cmd/cli/cmd/project.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/project.go]: Do we have the correct command prefix passed in?  I thought we didn't, so 'osc' isn't safe to hardcode.

----- Original Message -----

File: pkg/cmd/cli/cmd/project.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/project.go]: The name is generated by us vs named by the user.  I'll change it to 'contextNameIsGenerated'

----- Original Message -----

File: pkg/cmd/cli/cmd/project.go (hunk 3 above)
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/project.go]: Users already know who they are (they logged in as themselves) so the information in the context isn't relevant.  For normal happy path they don't care about that, so including it confuses them (thinks they did something wrong).  Reward the user with simple terse statements when they do the right thing (keep their mental model), explain to them when they step out of bounds by being more verbose.  That means they recognize "tons of gorp" as "Hrm, I did something wrong so now I need to read what they sent".

----- Original Message -----

**Comment by smarterclayton** [comment]
comments addressed [merge] or yell

**Pull request merged**